Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -49,15 +51,24 @@ private extension CardPresentPaymentStore {
         // Over simplification. This is the point where we would receive
         // new data via the CardReaderService's stream of discovered readers
         // In here, we should redirect that data to Storage and also up to the UI.
-        // For now we are sending the data up to the UI after mapping CardReaderService.CardReader
-        // to Yosemite.CardReader.
-        cancellable = cardReaderService.discoveredReaders.sink { readers in
-            let yosemiteReaaders = readers.map {
-                Yosemite.CardReader(name: $0.name, serialNumber: $0.serial)
-            }
-
-            // This hurts a bit, but for now it works
-            completion(yosemiteReaaders)
-        }
+        // For now we are sending the data up to the UI directly
+        cardReaderService.discoveredReaders.sink { readers in
+            completion(readers)
+        }.store(in: &cancellables)
+    }
+
+    func connect(reader: Yosemite.CardReader, onCompletion: @escaping (Result<[Yosemite.CardReader], Error>) -> Void) {
+        // We tiptoe around this for now. We will get into error handling later:
+        // https://github.com/woocommerce/woocommerce-ios/issues/3734
+        // https://github.com/woocommerce/woocommerce-ios/issues/3741
+        cardReaderService.connect(reader).sink(receiveCompletion: { error in
+        }, receiveValue: { (result) in
+        }).store(in: &cancellables)
+
+        // Dispatch completion block everytime the service published a new
+        // collection of connected readers
+        cardReaderService.connectedReaders.sink { connectedHardwareReaders in
```

Comment:
I _think_ (and I'm asking Stripe to confirm this), that there can only be zero or one connected reader... ever. If that is confirmed, I think we should have connectedReader and not an array.

Reply:
Sure. But we made it an array on purpose when we discussed the API of the service. Do you recall why?

Reply:
I think that might be my fault. I vaguely recall thinking that multiple P400s could be connected. Pretty sure now that they cannot be.